src/hotspot/share/interpreter/bytecodeTracer.cpp

```diff
@@ -472,19 +353,29 @@ void BytecodePrinter::print_attributes(int bci, outputStream* st) {
       st->print_cr(" " INT32_FORMAT, get_short());
       break;
     case Bytecodes::_ldc:
-      if (Bytecodes::uses_cp_cache(raw_code())) {
-        print_constant(get_index_u1_cpcache(), st);
-      } else {
-        print_constant(get_index_u1(), st);
+      {
+        int cp_index;
+        if (Bytecodes::uses_cp_cache(raw_code())) {
+          int obj_index = get_index_u1();
+          cp_index = constants()->object_to_cp_index(obj_index);
+        } else {
+          cp_index = get_index_u1();
+        }
+        print_constant(cp_index, st);
       }
       break;
 
     case Bytecodes::_ldc_w:
     case Bytecodes::_ldc2_w:
-      if (Bytecodes::uses_cp_cache(raw_code())) {
-        print_constant(get_index_u2_cpcache(), st);
-      } else {
-        print_constant(get_index_u2(), st);
+      {
+        int cp_index;
+        if (Bytecodes::uses_cp_cache(raw_code())) {
```

Comment:
Should this be uses_cp_cache() and is_linked()?

Reply:
I added an assert. Such bytecodes can appear only in linked classes.